### DISCUSSION TITLE
Read Files

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#805] [Docs]
<meta sectionIndex="3">
<meta url="https://github.com/johnlindquist/kit/discussions/805">
<meta id="D_kwDOEu7MBc4AP9Tb">
<meta title="Read Files">
<meta section="Files and Data">
<meta i="0">    
<meta path="docs/read-files">

## Read Files

```js
// Name: Read Files

import "@johnlindquist/kit"

let pkg = await readFile(
  home(".kenv", "package.json"),
  "utf-8"
)

await editor({
  value: pkg,
  language: "json",
})
```

[Open read-files in Script Kit](https://scriptkit.com/api/new?name=read-files&url=https://gist.githubusercontent.com/johnlindquist/00093125f3735bed8062dce0e20af4f8/raw/faec10fc8f99cc0744e9e840cfadb6cbe88e3a32/read-files.js")

**Comment by ekaone** [comment]
Hi John, thanks for this amazing tool, like it
can you show me how to read dir in Win 11, for example, my dev projects available on dir D:\, how I create this script?, thanks a lot

**Comment by josephcrawfordSRH** [comment]
@johnlindquist Is there a way to first check and see if the file exists?  I am trying to read a file that may not yet exist (cache file) and I am getting errors rather than an empty response.

**Reply by johnlindquist** [reply]
@josephcrawfordSRH 

First, please try the 2.0 release candidate:

https://github.com/johnlindquist/kitapp/releases/tag/v1.99.79



```js

let filePath = home("dev", "kit", "package.json")
let fileExists = await isFile(filePath) // There's also "isDir" and "isBin"
if (fileExists) {
  let data = await readJson(filePath)
  dev(data)
} else {
  throw Error("File does not exist")
}
```

**Reply by josephcrawfordSRH** [reply]
@johnlindquist thank you I will try to update on Monday and rework some code.
